Mage.Sets/src/mage/cards/t/TadeasJuniperAscendant.java

```diff
@@ -103,92 +101,38 @@ public boolean apply(Game game, Ability source) {
             return false;
         }
         permanent.untap(game);
-        game.addEffect(new TadeasJuniperAscendantBlockEffect(permanent, game), source);
+        game.addEffect(new TadeasJuniperAscendantEvasionEffect(permanent, game), source);
         return true;
     }
 }
 
-class TadeasJuniperAscendantBlockEffect extends RestrictionEffect {
+class TadeasJuniperAscendantEvasionEffect extends RestrictionEffect {
 
     private final MageObjectReference mor;
-
-    TadeasJuniperAscendantBlockEffect(Permanent permanent, Game game) {
-        super(Duration.EndOfTurn);
+    TadeasJuniperAscendantEvasionEffect(Permanent permanent, Game game) {
+        super(Duration.EndOfCombat);
         this.mor = new MageObjectReference(permanent, game);
+        staticText = "and can't be blocked by creatures with greater power";
     }
 
-    private TadeasJuniperAscendantBlockEffect(final TadeasJuniperAscendantBlockEffect effect) {
+    private TadeasJuniperAscendantEvasionEffect(final TadeasJuniperAscendantEvasionEffect effect) {
         super(effect);
         this.mor = effect.mor;
     }
 
-    @Override
-    public TadeasJuniperAscendantBlockEffect copy() {
-        return new TadeasJuniperAscendantBlockEffect(this);
-    }
-
-    @Override
-    public boolean canBeBlocked(Permanent attacker, Permanent blocker, Ability source, Game game, boolean canUseChooseDialogs) {
-        return false;
-    }
-
     @Override
     public boolean applies(Permanent permanent, Ability source, Game game) {
         Permanent attacker = mor.getPermanent(game);
-        if (attacker == null) {
-            discard();
-            return false;
-        }
-        return permanent.getPower().getValue() > attacker.getPower().getValue();
-    }
-}
-
-class TadeasJuniperAscendantTriggeredAbility extends TriggeredAbilityImpl {
-
-    private final Set<UUID> damagedPlayerIds = new HashSet<>();
-
-    TadeasJuniperAscendantTriggeredAbility() {
-        super(Zone.BATTLEFIELD, new DrawCardSourceControllerEffect(1), false);
-        this.withFlavorWord("Fierce Punch");
-        setTriggerPhrase("Whenever one or more creatures you control deal combat damage to a player, ");
+        if (attacker == null) return false;
+        return permanent == attacker;
```

Comment:
Equality operator `==` isn't appropriate for comparing objects like this. Need to check that the permanent and attacker have the same UUID and the same zcc (zone change counter, because the permanent isn't the same object if it leaves the battlefield and returns). The engine uses `MageObjectReference` to track a UUID along with a zcc. So you need to compare the stored MageObjectReference to a new MageObjectReference like so:

```
return mor.equals(new MageObjectReference(permanent, game));
```

... now that I think about it, if the target pointer is getting passed through, another way to do this would be to not have a MageObjectReference at all and use:

```
return this.targetPointer.getTargets(game, source).contains(permanent.getId());
```

I'll let you test and confirm.